lesion_analysis/time_point_lesion_evolution.py

```diff
@@ -0,0 +1,417 @@
+                        help='Path to the second image')
+    parser.add_argument('-seg-1', '--segmentation-first_image', required=True,
+                        help='Path to the lesion segmentation of the first image')
+    parser.add_argument('-seg-2', '--segmentation-second_image', required=True,
+                        help='Path to the lesion segmentation of the second image')
+    parser.add_argument('-o', '--output_folder', required=True,
+                        help='Path to the output folder')
+    parser.add_argument('--show', action='store_true',
+                        help='Whether to show the results or not')
+    return parser
+
+
+def perform_registration(path_first_image, path_second_image, path_first_lesion_seg, path_second_lesion_seg, path_output_folder):
+    """
+    This function performs the registration of the images and the segmentations using spinal cord toolbox
+
+    Args:
+        path_first_image: path to the first image
+        path_second_image: path to the second image
+        path_first_lesion_seg: path to the lesion segmentation of the first image
+        path_second_lesion_seg: path to the lesion segmentation of the second image
+        path_output_folder: path to the output folder
+    
+    Returns:
+        path_warp: path to the warp file
+        path_warpinv: path to the inverse warp file
+        path_second_image_registered: path to the second image registered
+        path_second_lesion_seg_registered: path to the second lesion segmentation registered
+    """
+
+    #first we segment the spinal cord on the two images using sct_deepseg_sc
+    os.system('sct_deepseg_sc -i ' + path_first_image + ' -c t2 -o ' + path_output_folder + '/first_image_sc_seg.nii.gz')
+    os.system('sct_deepseg_sc -i ' + path_second_image + ' -c t2 -o' + path_output_folder + '/second_image_sc_seg.nii.gz')
+
+    #then we compute the vertebral levels of the images using sct_label_vertebrae
+    os.system('sct_label_vertebrae -i ' + path_first_image + ' -s ' + path_output_folder + '/first_image_sc_seg.nii.gz' + ' -c t2 -ofolder ' + path_output_folder)
+    os.system('sct_label_vertebrae -i ' + path_second_image + ' -s ' + path_output_folder + '/second_image_sc_seg.nii.gz' + ' -c t2 -ofolder ' + path_output_folder)
+
+    #we remove labels that are not in common in both images
+    ##we first get the labels of the two images
+    first_label = nib.load(path_output_folder + '/first_image_sc_seg_labeled.nii.gz')
+    second_label = nib.load(path_output_folder + '/second_image_sc_seg_labeled.nii.gz')
+    first_label_data = first_label.get_fdata()
+    second_label_data = second_label.get_fdata()
+    common_labels = np.intersect1d(np.unique(first_label_data), np.unique(second_label_data))
+
+    #we then remove the labels that are not in common
+    for label in np.unique(first_label_data):
+        if label not in common_labels:
+            first_label_data[first_label_data == label] = 0
```

Comment:
you might want to consider some features of `sct_label_utils -remove-reference`

Reply:
I mentioned an issue with this in this [comment](https://github.com/ivadomed/canproco/issues/38#issuecomment-1673920787) about very large computation time.